graph/db/models/channel_auth_proof.go

```diff
@@ -1,35 +1,119 @@
 package models
 
+import (
+	"github.com/lightningnetwork/lnd/fn/v2"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
 // ChannelAuthProof is the authentication proof (the signature portion) for a
-// channel. Using the four signatures contained in the struct, and some
+// channel.
+//
+// For v1 channels:
+// Using the four node and bitcoin signatures contained in the struct, and some
 // auxiliary knowledge (the funding script, node identities, and outpoint) nodes
 // on the network are able to validate the authenticity and existence of a
 // channel. Each of these signatures signs the following digest: chanID ||
 // nodeID1 || nodeID2 || bitcoinKey1|| bitcoinKey2 || 2-byte-feature-len ||
 // features.
+//
+// For v2 channels:
+// The single schnorr signature signs the tlv fields of the v2 channel
+// announcement message which are in the signed range.
 type ChannelAuthProof struct {
+	// Version is the version of the channel announcement.
+	Version lnwire.GossipVersion
+
 	// NodeSig1Bytes are the raw bytes of the first node signature encoded
 	// in DER format.
-	NodeSig1Bytes []byte
+	//
+	// NOTE: v1 channel announcements only.
+	NodeSig1Bytes fn.Option[[]byte]
 
 	// NodeSig2Bytes are the raw bytes of the second node signature
 	// encoded in DER format.
-	NodeSig2Bytes []byte
+	//
+	// NOTE: v1 channel announcements only.
+	NodeSig2Bytes fn.Option[[]byte]
 
 	// BitcoinSig1Bytes are the raw bytes of the first bitcoin signature
 	// encoded in DER format.
-	BitcoinSig1Bytes []byte
+	//
+	// NOTE: v1 channel announcements only.
+	BitcoinSig1Bytes fn.Option[[]byte]
 
 	// BitcoinSig2Bytes are the raw bytes of the second bitcoin signature
 	// encoded in DER format.
-	BitcoinSig2Bytes []byte
+	//
+	// NOTE: v1 channel announcements only.
+	BitcoinSig2Bytes fn.Option[[]byte]
+
+	// Signature is the raw bytes of the single schnorr signature for v2
+	// channel announcements.
+	//
+	// NOTE: v2 channel announcements only.
+	Signature fn.Option[[]byte]
 }
 
-// IsEmpty check is the authentication proof is empty Proof is empty if at
-// least one of the signatures are equal to nil.
+// IsEmpty check is the authentication proof is empty Proof is empty.
 func (c *ChannelAuthProof) IsEmpty() bool {
-	return len(c.NodeSig1Bytes) == 0 ||
-		len(c.NodeSig2Bytes) == 0 ||
-		len(c.BitcoinSig1Bytes) == 0 ||
-		len(c.BitcoinSig2Bytes) == 0
+	// For v2 channel announcements, we only have a single signature.
+	if c.Signature.IsSome() {
+		return len(c.Signature.UnwrapOr([]byte{})) == 0
+	}
+
+	// For v1 channel announcements, we either have all four signatures or
+	// none.
+	return len(c.NodeSig1Bytes.UnwrapOr([]byte{})) == 0
```

Comment:
nit: this seems to defeat a bit the purpose of the optional fields, perhaps one could switch on the `Version` and return `IsNone`?